packages/world-module-erc20-own-store/test/MUDERC20.t.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import { Test } from "forge-std/Test.sol";
+import { console2 } from "forge-std/Console2.sol";
+import { GasReporter } from "@latticexyz/gas-report/src/GasReporter.sol";
+
+import { MUDERC20 } from "../src/MUDERC20.sol";
+import { Token } from "../src/codegen/tables/Token.sol";
+import { Balances } from "../src/codegen/tables/Balances.sol";
+import { Allowances } from "../src/codegen/tables/Allowances.sol";
+
+contract MUDERC20Test is Test, GasReporter {
+  MUDERC20 muderc20;
+  address alice = address(0x123);
+  address charlie = address(0x456);
+
+  function setUp() public {
+    muderc20 = new MUDERC20("MUDERC20", "MUD", 18);
+  }
+
+  function testMUDERC20SetUp() public {
+    startGasReport("MUDERC20 constructor");
+
+    assertTrue(address(muderc20) != address(0));
+
+    assertEq(Token.getDecimals(), 18);
+    assertEq(Token.getTotalSupply(), 0);
+    assertEq(Token.getName(), "MUDERC20");
+    assertEq(Token.getSymbol(), "MUD");
+
+    endGasReport();
+  }
+
+  function testMUDERC20Mint() public {
+    startGasReport("MUDERC20 mint");
```

Comment:
we should wrap the gas report around the thing we want gas measurement for (i.e. just the `mint` call here)

Reply:
Got it, thanks!